Code/Framework/AzCore/AzCore/Memory/HphaSchema.cpp

```diff
@@ -455,7 +450,6 @@ namespace AZ
         };
         static constexpr size_t DEBUG_UNKNOWN_SIZE = (size_t)-1;
 
-#ifdef DEBUG_ALLOCATOR
```

Comment:
This adds 128 bytes to the size of `HpAllocator` when `DebugAllocatorEnable == false`, due to the new `debug_record_map` (line 562), `AZStd::mutex` (line 564), and `size_t[2]` (line 567) members, which are not used when `DebugAllocatorEnable` is `false`. I'd rather see that debug functionality moved out into its own class, so that we're not impacting the `HpAllocator` when `DebugAllocatorEnable` is `false`.

Reply:
Well there is only 1 HpAllocator in the entire application.
So there 128 bytes total of extra memory used which I think is an acceptable tradeoff.